src/nvim/macros.h

```diff
@@ -226,4 +226,10 @@
 
 #define EMPTY_POS(a) ((a).lnum == 0 && (a).col == 0 && (a).coladd == 0)
 
+// Mark unused variables with _UNUSED, so that we can suppress warnings from
+// gcc's -Wunused-value and -Wunused-but-set-variable.
+// Prefix with _ to not clash with xdiff/xinclude.h which defines UNUSED for use
+// with function arguments.
+#define _UNUSED(x) ((void)(x))
```

Comment:
wouldn't a macro for assert be better that expands to `_UNUSED` `ifndef NDEBUG`. Doesn't nvim already have this?

Reply:
I couldn't find anything but I can have another look.

Reply:
We just have code like:

```c
#ifndef NDEBUG
  assert(foo);
#else
  (void)foo;
#endif
```
or, when possible, putting the entire block of code inside the `NDEBUG` check.